src/wp-includes/class-wp-theme.php

```diff
@@ -1993,16 +2001,39 @@ private function set_pattern_cache( array $patterns ) {
 			'version'  => $this->get( 'Version' ),
 			'patterns' => $patterns,
 		);
-		wp_cache_set( 'wp_theme_patterns_' . $this->stylesheet, $pattern_data, 'theme_files' );
+
+		/**
+		 * Filters whether cache theme files persistently.
+		 *
+		 * @since 6.6.0
+		 *
+		 * @param bool   $cache_expiration Expiration time for the cache. Setting a value to `false` would bypass caching. Default `WP_Theme::$cache_expiration`.
+		 * @param string $group            The caching group.
+		 * @param string $context          Additional context for better cache control.
+		 */
+		$cache_expiration = apply_filters( 'wp_cache_theme_files_persistently', self::$cache_expiration, 'theme_files', 'theme_block_patterns' );
```

Comment:
Couple small suggestions, I don't think we need to pass the cache group here because the filter name itself implies that this should only be used for things being cached to the `theme_files` group.

```suggestion
		/**
		 * Filters whether to cache theme files persistently.
		 *
		 * @since 6.6.0
		 *
		 * @param bool   $cache_expiration Expiration time for the cache. Setting a value to `false` would bypass caching. Default `WP_Theme::$cache_expiration`.
		 * @param string $context          Additional context for better cache control.
		 */
		$cache_expiration = apply_filters( 'wp_cache_theme_files_persistently', self::$cache_expiration, 'theme_block_patterns' );
```